eis_toolkit/prediction/mlp.py

```diff
@@ -0,0 +1,301 @@
+        loss_function: Loss function to be used. Defaults to 'binary_crossentropy'.
+        dropout_rate: Fraction of the input units to drop. Value must be >= 0 and <= 1. Defaults to None.
+        early_stopping: Whether or not to use early stopping in training. Defaults to True.
+        es_patience: Number of epochs with no improvement after which training will be stopped. Defaults to 5.
+        metrics: Metrics to be evaluated by the model during training and testing. Defaults to ['accuracy'].
+        random_state: Seed for random number generation. Sets Python, Numpy and Tensorflow seeds to make
+            program deterministic. Defaults to None (random state / seed).
+
+    Returns:
+        Trained MLP model and training history.
+
+    Raises:
+        InvalidParameterValueException: Some of the numeric parameters have invalid values.
+        InvalidDataShapeException: Shape of X or y is invalid.
+    """
+    # 1. Check input data
+    _check_ML_model_data_input(X=X, y=y)
+    _check_MLP_inputs(
+        neurons=neurons,
+        validation_split=validation_split,
+        learning_rate=learning_rate,
+        dropout_rate=dropout_rate,
+        es_patience=es_patience,
+        batch_size=batch_size,
+        epochs=epochs,
+        output_neurons=output_neurons,
+        loss_function=loss_function,
+    )
+
+    if random_state is not None:
+        keras.utils.set_random_seed(random_state)
+
+    # 2. Create and compile a sequential model
+    model = keras.Sequential()
+
+    for neuron in neurons:
+        model.add(keras.layers.Dense(units=neuron, activation=activation))
+
+        if dropout_rate is not None:
+            model.add(keras.layers.Dropout(dropout_rate))
+
+    model.add(keras.layers.Dense(units=output_neurons, activation=last_activation))
+
+    model.compile(
+        optimizer=_keras_optimizer(optimizer, learning_rate=learning_rate), loss=loss_function, metrics=metrics
+    )
+
+    # 3. Train the model
+    # Early stopping callback
+    callbacks = [keras.callbacks.EarlyStopping(monitor="val_loss", patience=es_patience)] if early_stopping else []
+
+    history = model.fit(
+        X,
+        y,
+        epochs=epochs,
+        validation_split=validation_split if validation_split else 0,
+        validation_data=validation_data,
+        batch_size=batch_size,
+        callbacks=callbacks,
+    )
+
+    return model, history.history
+
+
+@beartype
+def train_MLP_regressor(
+    X: np.ndarray,
+    y: np.ndarray,
+    neurons: Sequence[int],
+    validation_split: Optional[float] = 0.2,
+    validation_data: Optional[Tuple[np.ndarray, np.ndarray]] = None,
+    activation: Literal["relu", "linear", "sigmoid", "tanh"] = "relu",
+    output_neurons: int = 1,
+    last_activation: Literal["linear"] = "linear",
+    epochs: int = 50,
+    batch_size: int = 32,
+    optimizer: Literal["adam", "adagrad", "rmsprop", "sdg"] = "adam",
+    learning_rate: Number = 0.001,
```

Comment:
Since it can be misleading/confusing if the default `learning_rate` is substituted silently in the background if `sdg` is chosen as the optimizer, I think we could leave it like this. I suppose it is reasonable to assume that if the user changes the optimizer, they would also tinker other parameters and have an understanding of what they're doing?